### PR TITLE
Feature/inbox api/rudolph

### LINKED ISSUE #79: feat: [backend] Implement simulated inbox API for UC-01
## Summary
Implement the backend simulated inbox list and email detail APIs for UC-01, including trainee filtering and email-open interaction tracking.

## Problem / Value
UC-01 requires a trainee to view simulated emails and open a specific email. The backend must return only the authenticated trainee's assigned emails and record the open interaction.

## Proposed Solution
Implement `GET /simulations/inbox`, `GET /simulations/emails/:emailId`, and `POST /simulations/emails/:emailId/interactions` backed by Prisma and protected by auth middleware. Record `EMAIL_OPENED` when an email is opened and allow `LINK_CLICKED` only for safe simulated link interactions.

**Blocks:** #82 and #83.
**Suggested branch name:** `feature/inbox-api/rudolph`
**Story Points:** 3
**Support/Reviewer:** @FJNel or @Adri4no098

## Scope
Included:
- `GET /simulations/inbox`.
- `GET /simulations/emails/:emailId`.
- `POST /simulations/emails/:emailId/interactions` with `eventType` limited to `EMAIL_OPENED` or `LINK_CLICKED`.
- Authenticated-user filtering.
- Missing/unauthorised email handling.

Out of scope:
- Real email integration.
- Admin email campaign creation.
- Email sending.

## Files / Areas to Touch
- `apps/backend/src/`
- Inbox/simulations routes and services
- Interaction event service/model
- Prisma schema/client usage
- Shared inbox DTOs
- Backend unit/API tests in `apps/backend/tests/unit/**/*.test.ts`

## Tasks
- [ ] Implement `GET /simulations/inbox`.
- [ ] Implement `GET /simulations/emails/:emailId`.
- [ ] Implement `POST /simulations/emails/:emailId/interactions`.
- [ ] Filter all email queries by authenticated trainee.
- [ ] Record `EMAIL_OPENED` interaction.
- [ ] Add Vitest/Supertest unit/API tests for success, empty list, missing email, and cross-user access.
- [ ] Return DTOs that match shared types.

## Deadlines
First usable draft: 13 May
Final review-ready update: 17 May

## Acceptance Criteria
- [ ] Authenticated trainee only sees their own simulated emails.
- [ ] Email detail does not expose another user's email.
- [ ] Opening an email records an `EMAIL_OPENED` event.
- [ ] `POST /simulations/emails/:emailId/interactions` rejects invalid event types with a validation error.
- [ ] Missing emails return a safe 404.
- [ ] Tests cover authorised and unauthorised access.
- [ ] Unit/API tests live under `apps/backend/tests/unit/`.

### PR DESCRIPTION
**Pull request by RudolphLamp**
## Summary

Implemented the backend simulated inbox and email detail APIs for **UC-01**. This enables trainees to view their assigned simulated emails, retrieve detailed email content, and track interactions.

**Key changes:**
- Added `GET /simulations/inbox` to list emails for the authenticated trainee.
- Added `GET /simulations/emails/:emailId` for detailed email views.
- Added `POST /simulations/emails/:emailId/interactions` to track `EMAIL_OPENED` and `LINK_CLICKED` events.
- Implemented automatic interaction tracking: fetching email details for the first time automatically records an `EMAIL_OPENED` event.
- Localized test data and simulation logic for the South African context (e.g., **ABSA Bank**).
- Extended `validateRequest` middleware to support `validateParams` for type-safe UUID and parameter validation.

## Linked Issue

Closes #79

## Type of change

- [x] Feature
- [ ] Bug Fix
- [ ] Documentation
- [ ] Chore/Maintenance

## Testing

I verified the changes using Vitest and Supertest:
- **Unit/API Tests**: Created `apps/backend/tests/unit/simulation.api.test.ts` covering:
    - Successful inbox retrieval.
    - Authorized and unauthorized (cross-user) email access.
    - Interaction recording via both the POST endpoint and automatic GET-based tracking.
    - Validation error handling for invalid event types.
- **Environment**: Verified locally using `pnpm --filter @insightful-phish/backend test:unit`. All 30 tests passed.

## Review Notes

Reviewers should focus on:
- **`simulation.service.ts`**: The `$transaction` logic for atomicity when marking an email as read and recording the interaction event.
- **`validateRequest.ts`**: The addition of `validateParams` which ensures UUIDs are valid before reaching the controller.
- **Security**: The check in `getSimulatedEmail` and `recordEmailInteraction` that verifies the `ownerUserId` of the inbox matches the authenticated trainee's ID.

## Checklist

- [x] I tested the change locally
- [x] No unrelated changes are included
- [x] Relevant tests were added or updated if applicable
- [x] Documentation was updated if needed
- [x] No secrets, credentials, or sensitive values were committed

### CODE REVIEW

File: apps/backend/src/routes/simulation.routes.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  recordSimulatedEmailInteractionRequestSchema,
+} from '@insightful-phish/shared';
+import { Router } from 'express';
+import { getEmail, getInbox, recordInteraction } from '../controllers/simulation.controller.js';
+import { requireAuth } from '../middleware/requireAuth.js';
+import { validateBody, validateParams } from '../middleware/validateRequest.js';
+
+export const simulationRouter = Router();
+
+simulationRouter.get('/simulations/inbox', requireAuth, getInbox);
```
[the remark this answers is not in the document]
Comment (RudolphLamp) [reply apps/backend/src/routes/simulation.routes.ts]: okay buddy

File: apps/backend/src/routes/simulation.routes.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+simulationRouter.get(
+  '/simulations/emails/:emailId',
+  requireAuth,
```
[the remark this answers is not in the document]
Comment (RudolphLamp) [reply apps/backend/src/routes/simulation.routes.ts]: okay buddy

File: apps/backend/src/routes/simulation.routes.ts [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+  validateParams(getSimulatedEmailRequestParamsSchema),
+  getEmail,
+);
+
+simulationRouter.post(
+  '/simulations/emails/:emailId/interactions',
+  requireAuth,
```
[the remark this answers is not in the document]
Comment (RudolphLamp) [reply apps/backend/src/routes/simulation.routes.ts]: okay buddy

**Review by FJNel — CHANGES_REQUESTED** [review]
## Changes requested: align with the revised modular Demo 1 domain model

This PR implements backend UC-01 simulated inbox and email detail APIs with interaction tracking, but it must be aligned with the revised modular campaign model before merge.

### Summary of the PR intent
- Implements endpoints: `GET /simulations/inbox`, `GET /simulations/emails/:emailId`, `POST /simulations/emails/:emailId/interactions`.
- Changes backend simulation controller/routes/service/tests and shared validation middleware.

### Blocking domain-model conflicts
- Uses `prisma.simulatedInbox.findUnique({ where: { ownerUserId: userId } })`, conflicting with simulation-based inbox.
- Authorizes email access by `email.inbox.ownerUserId` instead of campaign assignment/item.
- Uses old event names: `EMAIL_OPENED`, `LINK_CLICKED`.
- Exposes `simulationContext.isPhishing` based on `PHISHING_EMAIL`, risking answer leakage.
- Returns `recommendedTrainingDocumentId`, possibly implying non-campaign sequencing.
- Records interaction events with `userId` and `simulatedEmailId` direct fields, not campaign context.
- Does not implement `EmailClassificationResponse`.

### Please remove
- All `ownerUserId` simulated inbox access.
- `PHISHING_EMAIL` simulation type logic.
- Trainee-facing `simulationContext.isPhishing` from pre-classification email detail.
- Old event names.

### Please change
- Change inbox access to campaign item based:
  - `GET /trainee/campaign-items/:campaignItemId/simulated-inbox`
- Keep email detail endpoint only if it validates campaign assignment/item access:
  - `GET /trainee/simulated-emails/:emailId`
- Use `SIMULATED_EMAIL_OPENED`, `SIMULATED_EMAIL_LINK_CLICKED`, and `SIMULATED_EMAIL_CLASSIFIED` event names.
- Resolve all access through `CampaignAssignment -> Campaign -> CampaignItem -> Simulation -> SimulatedInbox -> SimulatedEmail`.

### Please add
- Add `POST /trainee/simulated-emails/:emailId/classification`.
- Create `EmailClassificationResponse` when a trainee classifies an email.
- Support selected red flags through the join model if available.
- Add tests for access control through campaign assignments/items.
- Add tests that expected classification/red flags are not leaked before classification.
- Add tests that classification creates `EmailClassificationResponse`.
- Add tests that interaction events do not store sensitive submitted values.

### What can stay
- Controller/service/route separation and parameter validation.
- Rate limiting if already accepted.
- The idea for cross-user/cross-access denial tests, but update the access model.

### Required tests/docs updates
- Update tests to mock campaign assignment/item access, not `ownerUserId`.
- Update docs to reflect campaign-based simulated inbox and email classification.

### Suggested acceptance before merge
- [ ] All `ownerUserId` and old event names are removed.
- [ ] All simulated inbox and email access is campaign assignment/item based.
- [ ] Email classification endpoint and `EmailClassificationResponse` creation are implemented.
- [ ] Tests prove access control and no answer leakage.

### Review focus
- Ensure the entire simulated inbox and email classification flow is campaign-based, does not leak answers or red flags, and does not preserve the old user-owned inbox or event names.
